src/coreclr/jit/codegenarmarch.cpp

```diff
@@ -1767,42 +1767,31 @@ void CodeGen::genCodeForIndir(GenTreeIndir* tree)
 
         // On arm64-v8.4+ we can use ldapur* instructions with acquire/release semantics to
         // avoid full memory barriers if address is contained and unscaled
-        bool hasRcpc2 = compiler->compOpportunisticallyDependsOn(InstructionSet_Rcpc2);
-
-        bool handledWithLdapur = false;
-        if (hasRcpc2 && !addrIsInReg && tree->Addr()->OperIs(GT_LEA) && !tree->HasIndex() && (tree->Scale() == 1) &&
-            emitter::emitIns_valid_imm_for_unscaled_ldst_offset(tree->Offset()))
-        {
-            if (ins == INS_ldrb)
-            {
-                ins               = INS_ldapurb;
-                handledWithLdapur = true;
-            }
-            else if (ins == INS_ldrh)
-            {
-                ins               = INS_ldapurh;
-                handledWithLdapur = true;
-            }
-            else if (ins == INS_ldr)
-            {
-                ins               = INS_ldapur;
-                handledWithLdapur = true;
-            }
-        }
+        bool canUseLdapur = compiler->compOpportunisticallyDependsOn(InstructionSet_Rcpc2) && !addrIsInReg && tree->Addr()->OperIs(GT_LEA) && !tree->HasIndex() && (tree->Scale() == 1) &&
+            emitter::emitIns_valid_imm_for_unscaled_ldst_offset(tree->Offset());
 
         if ((ins == INS_ldrb) && addrIsInReg)
         {
-            ins = hasRcpc ? INS_ldaprb : INS_ldarb;
+            // RCPC2 is supported -> ldapurb (ldaprb with IMM offset)
+            // RCPC1 is supported -> ldaprb
+            // Fallback:          -> ldarb
+            ins = canUseLdapur ? INS_ldapurb : hasRcpc ? INS_ldaprb : INS_ldarb;
```

Comment:
`canUseLdapur` is only true if `!addrIsInReg` is true, but this code only runs if `addrIsInReg` is true (so `!addrIsInReg` is false). So does the `canUseLdapur` case ever get hit?

Reply:
ah, my follow-up clean up is not correct (or rather - it disables ldapur), let me revert it

Reply:
reverted, thanks for noticing!